### PR TITLE
add falcon keyboard

### PR DESCRIPTION
**Pull request by yulei**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/matrix/falcon/rules.mk [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+MCU = atmega32u4
+
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
```
Comment (drashna) [inline keyboards/matrix/falcon/rules.mk:11]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/matrix/falcon/rules.mk [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality on B7 by default
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+MIDI_ENABLE = no            # MIDI support
+UNICODE_ENABLE = no         # Unicode
+AUDIO_ENABLE = no           # Audio output on port C6
+FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
+RGBLIGHT_ENABLE = yes       # Use RGB underglow light
```
Comment (drashna) [inline keyboards/matrix/falcon/rules.mk:25]:
```suggestion
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
RGBLIGHT_ENABLE = yes       # Use RGB underglow light
```

File: keyboards/matrix/falcon/config.h [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+/* disable debug print */
+//#define NO_DEBUG
+
+/* disable print */
+//#define NO_PRINT
+
+//rgb light setting
+#define RGBLED_NUM      4
+#define RGB_DI_PIN      D1
+#define RGBLIGHT_ANIMATIONS
```
Comment (drashna) [inline keyboards/matrix/falcon/config.h:54]: could you explicitly define which animations are enabled? 
```suggestion
#define RGBLIGHT_EFFECT_BREATHING
#define RGBLIGHT_EFFECT_RAINBOW_MOOD
#define RGBLIGHT_EFFECT_RAINBOW_SWIRL
#define RGBLIGHT_EFFECT_SNAKE
#define RGBLIGHT_EFFECT_KNIGHT
#define RGBLIGHT_EFFECT_CHRISTMAS
#define RGBLIGHT_EFFECT_STATIC_GRADIENT
#define RGBLIGHT_EFFECT_RGB_TEST
#define RGBLIGHT_EFFECT_ALTERNATING
#define RGBLIGHT_EFFECT_TWINKLE
```

File: keyboards/matrix/falcon/config.h [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+#define RGBLIGHT_HUE_STEP 8
+#define RGBLIGHT_SAT_STEP 8
+#define RGBLIGHT_VAL_STEP 8
+
+//pin setting
+#define CAPS_PIN            E2
```
Comment (drashna) [inline keyboards/matrix/falcon/config.h:60]:
```suggestion
#define LED_CAPS_LOCK_PIN E2
```

File: keyboards/matrix/falcon/falcon.c [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    setPinOutput(CHG_EN_PIN);
+    writePinHigh(CHG_EN_PIN);
+
+    // enable led power
+    setPinOutput(LED_POWER_PIN);
+    writePinHigh(LED_POWER_PIN);
+
+    // clear caps led
+    setPinOutput(CAPS_PIN);
+    writePinLow(CAPS_PIN);
```
Comment (drashna) [inline keyboards/matrix/falcon/falcon.c:31]:
```suggestion
```

File: keyboards/matrix/falcon/falcon.c [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+}
+
+void keyboard_post_init_user(void)
+{
+  // Customise these values to desired behaviour
+  debug_enable=true;
+  //debug_matrix=true;
+  //debug_keyboard=true;
+  //debug_mouse=true;
+}
```
Comment (drashna) [inline keyboards/matrix/falcon/falcon.c:50]:
```suggestion
```

File: keyboards/matrix/falcon/rules.mk [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+# Bootloader selection
+BOOTLOADER = atmel-dfu
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = lite # Virtual DIP switch configuration
+MOUSEKEY_ENABLE = yes        # Mouse keys
+EXTRAKEY_ENABLE = yes # Audio control and System control
+CONSOLE_ENABLE = yes		# Console for debug
```
Comment (drashna) [inline keyboards/matrix/falcon/rules.mk:13]:
```suggestion
CONSOLE_ENABLE = no       # Console for debug
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/matrix/falcon/readme.md [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+
+PCB using atmega32u4 for the Matrix Falcon
+
+Keyboard Maintainer: [astro](https://github.com/yulei)  
+Hardware Supported: Matrix Falcon
+Hardware Availability: Limited GB  
+
+Make example for this keyboard (after setting up your build environment):
+
+    make matrix/falcon:default
```
Comment (fauxpark) [inline keyboards/matrix/falcon/readme.md:11]:
```suggestion
* Keyboard Maintainer: [astro](https://github.com/yulei)
* Hardware Supported: Matrix Falcon
* Hardware Availability: Limited GB

Make example for this keyboard (after setting up your build environment):

    make matrix/falcon:default

Flashing example for this keyboard:

    make matrix/falcon:default:flash
```

File: keyboards/matrix/falcon/readme.md [hunk 9]
```diff
@@ -0,0 +1,17 @@
+# Falcon AVR
+
+PCB using atmega32u4 for the Matrix Falcon
+
+* Keyboard Maintainer: [astro](https://github.com/yulei)
+* Hardware Supported: Matrix Falcon
+* Hardware Availability: Limited GB
+
+Make example for this keyboard (after setting up your build environment):
```
Comment (drashna) [inline keyboards/matrix/falcon/readme.md:9]: Could you add instructions on how to enter the bootloader, to flash new firmware? 

Eg: 
```markup
## Bootloader

Enter the bootloader in 3 ways:

* **Bootmagic reset**: Hold down the key at (0,0) in the matrix (usually the top left key or Escape) and plug in the keyboard
* **Physical reset button**: Briefly press the button on the back of the PCB - some may have pads you must short instead
* **Keycode in layout**: Press the key mapped to `RESET` if it is available
```

**Review by fauxpark — APPROVED** [review]

**Review by tzarc — APPROVED** [review]

**Pull request merged**